### PR TITLE
Add support for CLIENT SETINFO

### PR DESCRIPTION
**Pull request by mgravell**
includes:

- ConfigurationOptions (to opt-opt)
- handshake (to send)
- release notes
- configuration documentation

note we can't validate this yet as not on any released servers

most contentious point: what lib-name to use - I've gone with `SE.Redis`, but: happy to use `StackExchange.Redis` if people prefer; I'm *not* aiming to make the name configurable

cross-reference: https://github.com/redis/redis/pull/11758

### CODE REVIEW

**Review by philon-msft — APPROVED** [review]

File: src/StackExchange.Redis/ClientInfo.cs [hunk 1]
```diff
@@ -180,6 +180,21 @@ public ClientType ClientType
             }
         }
 
+        /// <summary>
+        /// Client RESP protocol version. Added in Redis 7.0
+        /// </summary>
+        public string? ProtocolVersion { get; private set; }
+
+        /// <summary>
+        /// Client library version. Added in Redis 8.0
```
Comment (NickCraver) [inline src/StackExchange.Redis/ClientInfo.cs:189]:
```suggestion
        /// Client library name. Added in Redis 8.0
```

**Review by NickCraver — CHANGES_REQUESTED** [review]
Few tweaks then good to go!

File: src/StackExchange.Redis/ClientInfo.cs [hunk 2]
```diff
@@ -180,6 +180,21 @@ public ClientType ClientType
...
+        /// </summary>
+        public string? LibraryName { get; private set; }
+
+        /// <summary>
+        /// Client library version. Added in Redis 8.0
```
Comment (philon-msft) [inline src/StackExchange.Redis/ClientInfo.cs:194]: I believe the server will support the new SETINFO subcommand in Redis 7.2 (not 8.0)

Comment (NickCraver) [reply src/StackExchange.Redis/ClientInfo.cs]: +1 - fixed

**Review by NickCraver — APPROVED** [review]
:+1: Current looks good!

**Comment by NickCraver** [comment]
@philon-msft could you please double check me on latest?

**Comment by philon-msft** [comment]
> @philon-msft could you please double check me on latest?

Looks great to me @NickCraver

**Pull request merged**